### DISCUSSION TITLE
File Location to Add Playbook Field Mappings

### DISCUSSION THREAD

**Discussion by cyberphor** [Security-Onion-Solutions/securityonion#8616] [Unsupported Versions] [answered]
Hello, 

Where can I safely modify the file referenced below? I have field mappings I want to add. 

https://github.com/Security-Onion-Solutions/securityonion-image/blob/master/so-soctopus/so-soctopus/playbook/securityonion-baseline.yml

I have the following details to share:
* Version: 2.3.120
* Deployment: On-prem
* Install: ISO image

**Comment by defensivedepth** [comment, marked as answer]
This is not currently possible, but we are looking at a way to allow modifications.

In the meantime, what field mappings are you wanting to add?

**Reply by cyberphor** [reply]
OK. Trying to add "TargetUserName: winlog.event_data.TargetUserName."

**Reply by btdewey** [reply]
We have been using an ingest pipeline to create the field in the incoming data.  Not the most elegant solution, but it works.

**Reply by cyberphor** [reply]
i'm interested. can you share more on how you did this? did you use a Logstash parser?

**Comment by btdewey** [comment]
We copied the endgame logstash pipeline to the local directory structure
and then modified it.  We used if statements to match the specific data
types then used a mutate filter to add the specific field(s).  I don't see
why this also couldn't be done with an Elastic search pipeline if you
wanted to.  There is some additional extraction we are doing so doing all
the transformation in one place is less complex.

Bill

**Reply by cyberphor** [reply]
awesome, thanks!
